stubs/untangle/untangle.pyi

```diff
@@ -0,0 +1,37 @@
+from _typeshed import Incomplete
+from collections.abc import Iterator
+from typing import Any
+from xml.sax import handler
+
+def is_string(x: Any) -> bool: ...
+
+class Element:
+    children: Incomplete
+    is_root: bool
+    cdata: str
+    def __init__(self, name: str | None, attributes: dict[str, Any] | None) -> None: ...
+    def add_child(self, element: Element) -> None: ...
+    def add_cdata(self, cdata: str) -> None: ...
+    def get_attribute(self, key: str) -> Any: ...
+    def get_elements(self, name: Incomplete | None = ...) -> list[Element]: ...
+    def __getitem__(self, key: str) -> Any: ...
+    def __getattr__(self, key: str) -> Element: ...
+    def __hasattribute__(self, name: str) -> bool: ...
+    def __iter__(self) -> Iterator[Element]: ...
```

Comment:
If this method is called on a subclass of Element, it will in fact yield an instance of the subclass rather than an instance of `Element`. We can model this behaviour using a `TypeVar`. In typeshed, we prefer to use the same `TypeVar` every time for places where this happens: a special TypeVar that's called `Self`:

```suggestion
    def __iter__(self: Self) -> Iterator[Self]: ...
```

(You'll need to add `from _typeshed import Self` to the top of the file)